src/main/query_profiler.cpp

```diff
@@ -263,24 +268,36 @@ void QueryProfiler::EndPhase() {
 	}
 }
 
-OperatorProfiler::OperatorProfiler(ClientContext &context) {
-	enabled = QueryProfiler::Get(context).IsEnabled();
-	settings = ClientConfig::GetConfig(context).profiler_settings;
-}
-
-bool OperatorProfiler::SettingIsEnabled(MetricsType metric) const {
+bool SettingIsEnabled(profiler_settings_t settings, MetricsType metric) {
 	if (settings.find(metric) != settings.end()) {
 		return true;
 	}
-	if (metric == MetricsType::OPERATOR_TIMING && SettingIsEnabled(MetricsType::CPU_TIME)) {
-		return true;
-	}
-	if (metric == MetricsType::OPERATOR_CARDINALITY && SettingIsEnabled(MetricsType::CUMULATIVE_CARDINALITY)) {
-		return true;
+
+	switch (metric) {
+	case MetricsType::OPERATOR_TIMING:
+		return SettingIsEnabled(settings, MetricsType::CPU_TIME);
+	case MetricsType::OPERATOR_CARDINALITY:
+		return SettingIsEnabled(settings, MetricsType::CUMULATIVE_CARDINALITY);
+	case MetricsType::OPERATOR_ROWS_SCANNED:
+		return SettingIsEnabled(settings, MetricsType::CUMULATIVE_ROWS_SCANNED);
+	default:
+		break;
 	}
+
 	return false;
 }
 
+OperatorProfiler::OperatorProfiler(ClientContext &context) : context(context) {
```

Comment:
Are we now keeping two `settings`? Or a subset?

Reply:
We're now just keeping a subset of settings that are relevant to the operator. Since we don't have access to the profiling settings in `OperatorProfiler` we need to store at least a reference here.

Reply:
Makes sense 👍